src/transaction/log_recovery_redo_parallel.hpp

```diff
@@ -0,0 +1,274 @@
+/*
+ * Copyright 2008 Search Solution Corporation
+ * Copyright 2016 CUBRID Corporation
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ */
+
+#ifndef _LOG_RECOVERY_REDO_PARALLEL_HPP_
+#define _LOG_RECOVERY_REDO_PARALLEL_HPP_
+
+#include "log_compress.h"
```

Comment:
I think all headers are only SERVER_MODE.

Some headers may be replaced with forward declarations (thread related ones)

<bitset> is probably unused anymore.

Reply:
Yes, better to guard all of them `SERVER_MODE` just to be on the safe side.